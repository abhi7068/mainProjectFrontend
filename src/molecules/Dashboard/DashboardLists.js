import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import List from '@material-ui/core/List';
import DashboardIcon from '@material-ui/icons/Dashboard';
import ShoppingCartIcon from '@material-ui/icons/ShoppingCart';
import PeopleIcon from '@material-ui/icons/People';
import LayersIcon from '@material-ui/icons/Layers';
import Collapse from '@material-ui/core/Collapse';
import ExpandLess from '@material-ui/icons/ExpandLess';
import ExpandMore from '@material-ui/icons/ExpandMore';
import { withRouter, NavLink } from 'react-router-dom';
import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart';
import ListAltIcon from '@material-ui/icons/ListAlt';
import CategoryIcon from '@material-ui/icons/Category';
import PropTypes from 'prop-types';
import clsx from 'clsx';

import React from 'react';

const ListItems = (props) => {
  const { classes } = props;
  const [open, setpOpen] = React.useState(false);
  const [out, setcOpen] = React.useState(false);

  const handleProduct = () => {
    setpOpen(!open);
  };
  const handleClick = () => {
    setcOpen(!out);
  };

  return (
    <div>
      <NavLink
        exact
        style={{ color: 'white' }}
        activeClassName="is-active"
        activeStyle={{ color: 'red' }}
        to="/admin/dashboard"
      >
        <ListItem button>
          <ListItemIcon>
            <DashboardIcon
              className={classes.customHoverFocus}
              style={{ color: 'white' }}
            />
          </ListItemIcon>
          Dashboard
          {/* <ListItemText style={{ color: 'white' }} primary="Dashboard" /> */}
        </ListItem>
      </NavLink>

      <ListItem button onClick={handleProduct}>
        <ListItemIcon style={{ color: 'white' }}>
          <ShoppingCartIcon />
        </ListItemIcon>
        <ListItemText style={{ color: 'white' }} primary="Products" />
        {open ? (
          <ExpandLess style={{ color: 'white' }} />
        ) : (
          <ExpandMore style={{ color: 'white' }} />
        )}
      </ListItem>
      <Collapse in={open} timeout="auto" unmountOnExit>
        <List component="div" disablePadding>
          <NavLink
            data-testid="dashboard"
            exact
            style={{ color: 'white' }}
            activeClassName="is-active"
            activeStyle={{ color: 'red' }}
            to="/admin/addproduct"
          >
            <ListItem button className={clsx(classes.nested)}>
              <ListItemIcon style={{ color: 'white' }}>
                <AddShoppingCartIcon className={classes.size} />
              </ListItemIcon>
              Add Products
            </ListItem>
          </NavLink>
          <NavLink
            exact
            style={{ color: 'white' }}
            activeClassName="is-active"
            activeStyle={{ color: 'red' }}
            to="/admin/listproducts"
          >
            <ListItem button className={classes.nested}>
              <ListItemIcon style={{ color: 'white' }}>
                <ListAltIcon className={classes.size} />
              </ListItemIcon>
              List of Products
            </ListItem>
          </NavLink>
        </List>
      </Collapse>

      <ListItem button onClick={handleClick} style={{ color: 'white' }}>
        <ListItemIcon style={{ color: 'white' }}>
          <LayersIcon />
        </ListItemIcon>
        <ListItemText primary="Categories" />
        {out ? (
          <ExpandLess style={{ color: 'white' }} />
        ) : (
          <ExpandMore style={{ color: 'white' }} />
        )}
      </ListItem>
      <Collapse in={out} timeout="auto" unmountOnExit>
        <List component="div" disablePadding>
          <NavLink
            exact
            style={{ color: 'white' }}
            activeClassName="is-active"
            activeStyle={{ color: 'red' }}
            to="/admin/add-category"
          >
            <ListItem button className={classes.nested}>
              <ListItemIcon style={{ color: 'white' }}>
                <CategoryIcon className={classes.size} />
              </ListItemIcon>
              Add Categories
            </ListItem>
          </NavLink>
          <NavLink
            exact
            style={{ color: 'white' }}
            activeClassName="is-active"
            activeStyle={{ color: 'red' }}
            to="/admin/listcategories"
          >
            <ListItem
              button
              className={classes.nested}
              // style={{ color: 'white' }}
            >
              <ListItemIcon style={{ color: 'white' }}>
                <ListAltIcon className={classes.size} />
              </ListItemIcon>
              List of Categories
            </ListItem>
          </NavLink>
        </List>
      </Collapse>
      <NavLink
        exact
        style={{ color: 'white' }}
        activeClassName="is-active"
        activeStyle={{ color: 'red' }}
        to="/admin/list-users"
      >
        <ListItem
          button
          onClick={() => {
            props.history.push('/admin/list-users');
          }}
        >
          <ListItemIcon style={{ color: 'white' }}>
            <PeopleIcon />
          </ListItemIcon>
          Users
        </ListItem>
      </NavLink>
    </div>
  );
};
ListItems.propTypes = {
  history: PropTypes.object,
  classes: PropTypes.object,
};

export default withRouter(ListItems);
