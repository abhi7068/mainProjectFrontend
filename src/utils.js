import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
