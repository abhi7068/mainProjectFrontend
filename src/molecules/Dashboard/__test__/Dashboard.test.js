import React from 'react';
import { render } from '../../../utils';
import DashboardLists from '../DashboardLists';
/* import ReactDOM from 'react-dom'; */

it('it renders', () => {
  render(<DashboardLists classes="" />);
});
it('snapshot', () => {
  const { asFragment } = render(<DashboardLists classes="" />);
  expect(asFragment(<DashboardLists classes="" />)).toMatchSnapshot();
});
