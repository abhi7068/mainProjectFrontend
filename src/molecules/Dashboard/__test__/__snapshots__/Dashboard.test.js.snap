// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      href="/admin/dashboard"
      style="color: white;"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            style="color: white;"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
            />
          </svg>
        </div>
        Dashboard
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </a>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
        style="color: white;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root"
        style="color: white;"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Products
        </span>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        style="color: white;"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      style="color: white;"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root"
        style="color: white;"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 18.54l-7.37-5.73L3 14.07l9 7 9-7-1.63-1.27-7.38 5.74zM12 16l7.36-5.73L21 9l-9-7-9 7 1.63 1.27L12 16z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Categories
        </span>
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        style="color: white;"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <a
      href="/admin/list-users"
      style="color: white;"
    >
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root"
          style="color: white;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
        </div>
        Users
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </a>
  </div>
</DocumentFragment>
`;
